src/RenderingComponent.cpp

```diff
@@ -48,13 +77,24 @@ void RenderingComponent::set_mesh(MeshAsset* mesh) {
     setupBuffer();
 }
 
+void RenderingComponent::set_texture(TextureAsset* texture) {
+    texture_ = texture;
+}
+
+void RenderingComponent::set_shader(ShaderAsset* shader) {
+    shader_ = shader;
+}
+
 void RenderingComponent::setupBuffer() {
-    glGenBuffers(1, &gl_vertex_buffer_);
-    glBindBuffer(GL_ARRAY_BUFFER, gl_vertex_buffer_);
-    glBufferData(GL_ARRAY_BUFFER, sizeof(MeshAsset::VertexData)*mesh_->vertices_.size(), &mesh_->vertices_.front(), GL_STATIC_DRAW);
 
-    glGenBuffers(1, &gl_index_buffer_);
-    glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, gl_index_buffer_);
-    glBufferData(GL_ELEMENT_ARRAY_BUFFER, mesh_->indices_.size() * sizeof(GLuint), &mesh_->indices_.front(), GL_STATIC_DRAW);
+    if (mesh_ != nullptr && mesh_->valid_) {
```

Comment:
What if it's not? I feel like some error should be shown instead of nothing happening silently.